### PR TITLE
created head and theme logic

### LINKED ISSUE #2: Create a header
Create the Header layout based on design:

Image

Acceptance criterios: 
- light/dark theme switcher with correspondent icons
- box shadow on the bottom

All future components should support light/dark themes so we need a mechanism to share theme state between components.

### PR DESCRIPTION
**Pull request by LazarevSlava**
# Changed

- Header was added
- Logic for changing them was added

## Checklist

- [x] Linter checks
- [x] Code style checks

### CODE REVIEW

File: src/components/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import icon from "../assets/images/night.png";
+import style from "./Header.module.scss";
+import iconMoon from "../assets/images/moon.png";
```
Comment (michey85) [inline src/components/Header.jsx]: Is it possible to have the same images as SVG?

And I'm not sure about the naming. At least the first one shouldn't be just `icon`.

File: src/components/Header.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { useContext } from "react";
+import { CustomContext } from "../helpers/Context";
```
Comment (michey85) [inline src/components/Header.jsx]: Let's create a `contexts` folder. It isn't a regular helper.

File: src/components/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+function Header() {
+  const { isTheme = true, toggleTheme } = useContext(CustomContext);
```
Comment (michey85) [inline src/components/Header.jsx]: Please consider creating a custom hook for the specific context.

Reference https://kentcdodds.com/blog/how-to-use-react-context-effectively

File: src/components/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  return (
+    <>
+      {isTheme ? (
```
Comment (michey85) [inline src/components/Header.jsx]: Why `isTheme` is a boolean? Light theme is considered as non-theme? =)

File: src/components/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        <div className={style.head_block}>
+          <h1>Where in the world?</h1>
+          <h3 onClick={toggleTheme}>
+            <img src={icon} alt="moon"></img>
```
Comment (michey85) [inline src/components/Header.jsx]: Img tag has no closing tag, man.

File: src/components/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            Dark Mode
+          </h3>
+        </div>
+      ) : (
+        <div className={`${style.head_block} ${style.light}`}>
+          <h1>Where in the world?</h1>
```
Comment (michey85) [inline src/components/Header.jsx]: I think you repeat yourself with the markup.

I propose creating a different approach
```js
const C = () => {
  const containerClassnames = `${style.head_block}${isDarkTheme ? ` ${style.light}` : ''`}`;
  const imageAltText = isDarkTheme ? 'dark moon' : 'moon'; // actually this one may be the same text
  const themeText = isDarkTheme ? 'Light theme' : 'Dark theme'
  const ThemeIcon = isDarkTheme ? Icon : IconMoon;

  return <div className={containerClassnames}>
    <h1>Where in the world?</h1>
   <h3 onClick={toggleTheme}>
            <img src={ThemeIcon} alt={imageAltText} />
            {themeText}
    </h3>
 </div>
}
```

File: src/helpers/Context.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { createContext, useState } from "react";
+import PropTypes from "prop-types";
+
+export const CustomContext = createContext();
```
Comment (michey85) [inline src/helpers/Context.jsx]: The name itself says nothing about what is inside. How will you name your second context then?

File: src/components/Header.module.scss [hunk 8]
```diff
@@ -0,0 +1,33 @@
+@import "./utils/index.scss";
+.head_block {
+  @include theme($theme: $darkBlue);
+  transition:
+    background-color 0.8s,
+    color 0.8s;
```
Comment (michey85) [inline src/components/Header.module.scss]: When you use the same timing for different properties you don't need to write it each time.
```
transition: background-color, color 0.8s;
```

File: src/components/Header.jsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import { CustomContext } from "../helpers/Context";
+
+function Header() {
+  const { isTheme = true, toggleTheme } = useContext(CustomContext);
+  return (
+    <>
+      {isTheme ? (
+        <div className={style.head_block}>
+          <h1>Where in the world?</h1>
+          <h3 onClick={toggleTheme}>
```
Comment (michey85) [inline src/components/Header.jsx]: Why h3?

File: src/components/Header.module.scss [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  align-items: center;
+}
+h3 {
+  font-weight: normal;
+  cursor: pointer;
+}
+img {
+  padding-top: 1px;
+  width: 1em;
+  height: 1em;
```
Comment (michey85) [inline src/components/Header.module.scss]: Are you sure you wanna use `em`? 😃

File: src/components/Header.module.scss [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  margin-right: 0.5em;
+  filter: invert(100%) brightness(1000%);
```
Comment (michey85) [inline src/components/Header.module.scss]: Are you sure you need this filter?

**Review by michey85 — CHANGES_REQUESTED** [review]

File: src/components/Header.jsx (hunk 1 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: The problem is in template we see an empty moon and a full moon. I'm not sure I knew how to change them.
If you can give me a little advance  I will try
name changed :)

File: src/components/Header.jsx (hunk 2 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: ok

File: src/components/Header.jsx (hunk 1 above)
Comment (michey85) [reply src/components/Header.jsx]: I'm sure there are a lot of svg icons in the internet. There are libraries like react-icons, or websites like flaticon.com where you can find something similar and just grab svg code from there.

File: src/helpers/Context.jsx (hunk 7 above)
Comment (LazarevSlava) [reply src/helpers/Context.jsx]: ok

File: src/components/Header.module.scss (hunk 8 above)
Comment (LazarevSlava) [reply src/components/Header.module.scss]: ok

File: src/components/Header.jsx (hunk 9 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: )) why not? 
Really I don’t  know, better h2?

File: src/components/Header.module.scss (hunk 10 above)
Comment (LazarevSlava) [reply src/components/Header.module.scss]: ok rem

File: src/components/Header.module.scss (hunk 11 above)
Comment (LazarevSlava) [reply src/components/Header.module.scss]: yes, because  color: white doesn’t  work

Comment (michey85) [reply src/components/Header.module.scss]: This is why you need SVG instead of png.

File: src/components/Header.jsx (hunk 9 above)
Comment (michey85) [reply src/components/Header.jsx]: Well, first of all it doesn't look as a heading. It's just a span to me. There is no semantical meaning for the app.
Next, we should use headings without skipping them. If there is no h2 on a page h3 is a semantic error.

File: src/components/Header.module.scss (hunk 11 above)
Comment (michey85) [reply src/components/Header.module.scss]: With svg you can use any color you want https://www.youtube.com/watch?v=b4btk12Evgw

File: src/components/Header.jsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
+import style from "./Header.module.scss";
+import iconMoonNight from "../assets/images/half-moon-shape-svgrepo-com.svg";
```
Comment (michey85) [inline src/components/Header.jsx]: Use it as a component. With vite you need to find out how exactly you can achieve that.

File: src/components/Header.jsx (hunk 1 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: done

File: src/components/Header.jsx (hunk 4 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: done

File: src/components/Header.jsx (hunk 5 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: done

File: src/components/Header.module.scss (hunk 11 above)
Comment (LazarevSlava) [reply src/components/Header.module.scss]: I tried  a lot ^ but there is an issue  the I can’t  resolve

Comment (michey85) [reply src/components/Header.module.scss]: Which one?

Comment (LazarevSlava) [reply src/components/Header.module.scss]: Uncaught SyntaxError: The requested module '/src/assets/images/half-moon-shape-svgrepo-com.svg?import' does not provide an export named 'ReactComponent' (at Header.jsx:2:10)

File: src/components/Header.jsx (hunk 9 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: done

File: src/components/Header.jsx (hunk 12 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: done

File: src/components/Header.jsx (hunk 3 above)
Comment (LazarevSlava) [reply src/components/Header.jsx]: 2,5 hours my friend, 2,5 hours!!!

Comment (michey85) [reply src/components/Header.jsx]: Like a turtle =)

**Review by michey85 — APPROVED** [review]
LGTM

**Pull request merged**